src/python/grpcio_tests/tests_aio/unit/timeout_test.py

```diff
@@ -0,0 +1,178 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests behavior of the timeout mechanism on client side."""
+
+import asyncio
+import logging
+import platform
+import random
+import unittest
+import datetime
+
+import grpc
+from grpc.experimental import aio
+
+from tests_aio.unit._test_base import AioTestBase
+from tests_aio.unit import _common
+
+_SLEEP_TIME_UNIT_S = datetime.timedelta(seconds=1).total_seconds()
+
+_TEST_SLEEPY_UNARY_UNARY = '/test/Test/SleepyUnaryUnary'
+_TEST_SLEEPY_UNARY_STREAM = '/test/Test/SleepyUnaryStream'
+_TEST_SLEEPY_STREAM_UNARY = '/test/Test/SleepyStreamUnary'
+_TEST_SLEEPY_STREAM_STREAM = '/test/Test/SleepyStreamStream'
+
+_REQUEST = b'\x00\x00\x00'
+_RESPONSE = b'\x01\x01\x01'
+
+# logging.getLogger('asyncio').setLevel(logging.WARNING)
+
+class _GenericHandler(grpc.GenericRpcHandler):
+
+    def __init__(self):
+        self._routing_table = {
+            _TEST_SLEEPY_UNARY_UNARY:
+                grpc.unary_unary_rpc_method_handler(self._test_sleepy_unary_unary
+                                                   ),
+            _TEST_SLEEPY_UNARY_STREAM:
+                grpc.unary_stream_rpc_method_handler(self._test_sleepy_unary_stream),
+            _TEST_SLEEPY_STREAM_UNARY:
+                grpc.stream_unary_rpc_method_handler(self._test_sleepy_stream_unary),
+            _TEST_SLEEPY_STREAM_STREAM:
+                grpc.stream_stream_rpc_method_handler(self._test_sleepy_stream_stream)
+        }
+
+    @staticmethod
+    async def _test_sleepy_unary_unary(unused_request, unused_context):
+        await asyncio.sleep(_SLEEP_TIME_UNIT_S)
+        return _RESPONSE
+
+    @staticmethod
+    async def _test_sleepy_unary_stream(unused_request, unused_context):
+        yield _RESPONSE
+        await asyncio.sleep(_SLEEP_TIME_UNIT_S)
+        yield _RESPONSE
+
+    @staticmethod
+    async def _test_sleepy_stream_unary(unused_request_iterator, context):
+        assert _REQUEST == await context.read()
+        await asyncio.sleep(_SLEEP_TIME_UNIT_S)
+        assert _REQUEST == await context.read()
+        return _RESPONSE
+
+    @staticmethod
+    async def _test_sleepy_stream_stream(unused_request_iterator, context):
+        assert _REQUEST == await context.read()
+        await asyncio.sleep(_SLEEP_TIME_UNIT_S)
+        await context.write(_RESPONSE)
+
+    def service(self, handler_call_details):
+        return self._routing_table.get(handler_call_details.method)
+
+
+async def _start_test_server():
+    server = aio.server()
+    port = server.add_insecure_port('[::]:0')
+    server.add_generic_rpc_handlers((_GenericHandler(),))
+    await server.start()
+    return 'localhost:%d' % port, server
```

Comment:
Nit: f-string?

Reply:
Updated.

Once, our Kokoro machines are running Python 3.5, which if we are using f-string then it starts failing. It should be fixed by now (or it is still testing 3.5).